Content.Server/Silicons/Bots/MedibotSystem.cs

```diff
@@ -0,0 +1,175 @@
+
+            if (!_interactionSystem.InRangeUnobstructed(performer, target, 2f))
+                return false;
+
+            // Hold still, please
+            if (TryComp<PhysicsComponent>(target, out var physics) && physics.LinearVelocity.Length != 0f)
+                return false;
+
+            // Figure out which drug we're going to inject, if any.
+            if (!ChooseDrug(target, component, out var drug, out var injectAmount))
+            {
+                _popups.PopupEntity(Loc.GetString("medibot-cannot-inject"), performer, performer, PopupType.SmallCaution);
+                return false;
+            }
+
+            component.InjectTarget = target;
+
+            _popups.PopupEntity(Loc.GetString("medibot-inject-receiver", ("bot", performer)), target, target, PopupType.Medium);
+            _popups.PopupEntity(Loc.GetString("medibot-inject-actor", ("target", target)), performer, performer, PopupType.Medium);
+
+            var data = new MedibotInjectData(injectable, drug, injectAmount);
+            var args = new DoAfterEventArgs(performer, component.InjectDelay, target: target)
+            {
+                BreakOnTargetMove = true,
+                BreakOnUserMove = true,
+                BreakOnStun = true,
+                NeedHand = false
+            };
+
+            component.IsInjecting = true;
+            _doAfter.DoAfter(args, data);
+
+            return true;
+        }
+
+        private bool SharedInjectChecks(EntityUid uid, EntityUid target, [NotNullWhen(true)] out Solution? injectable)
+        {
+            injectable = null;
+            if (_mobs.IsDead(target))
+                return false;
+
+            if (HasComp<NPCRecentlyInjectedComponent>(target))
+                return false;
+
+            if (!_solution.TryGetInjectableSolution(target, out var injectableSol))
+                return false;
+
+            injectable = injectableSol;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Chooses which drug to inject based on info about the target.
+        /// With a small rewrite shouldn't be hard to make different kinds of medibots.
+        /// </summary>
+        private bool ChooseDrug(EntityUid target, MedibotComponent component, out string drug, out float injectAmount, DamageableComponent? damage = null)
+        {
+            drug = "None";
+            injectAmount = 0;
+            if (!Resolve(target, ref damage))
+                return false;
+
+            if (damage.TotalDamage == 0)
+                return false;
+
+            if (_mobs.IsCritical(target))
+            {
+                drug = component.EmergencyMed;
+                injectAmount = component.EmergencyMedInjectAmount;
+                return true;
+            }
+
+            if (damage.TotalDamage <= 50)
+            {
+                drug = component.StandardMed;
+                injectAmount = component.StandardMedInjectAmount;
+                return true;
+            }
```

Comment:
Due to this it feels a lot like having a built in injector should be rolled out into a separate system.